PiscadaServer/src/apps/ApiServer/documents/documentservices.cpp

```diff
@@ -0,0 +1,213 @@
+#include "documents/documentservices.h"
+#include "apiserverinfo.h"
+#include "apiserverdatabase.h"
+#include <QDebug>
+#include <QFile>
+#include <QDir>
+
+ApiError DocumentServices::decodeAndValidateBase64File(const QByteArray &fileContent, const QString &expectedExtension, QByteArray &decodedContent)
+{
+    const QByteArray headerPrefix = "data:";
+    int commaIndex = fileContent.indexOf(',');
+
+    if (fileContent.startsWith(headerPrefix) && commaIndex != -1)
+    {
+        QByteArray header = fileContent.left(commaIndex);
+        QByteArray base64Data = fileContent.mid(commaIndex + 1);
+        decodedContent = QByteArray::fromBase64(base64Data);
+
+        int semiIndex = header.indexOf(';');
+        QString mimeType;
+        if (semiIndex != -1)
+        {
+            mimeType = QString::fromUtf8(header.mid(headerPrefix.length(), semiIndex - headerPrefix.length()));
+        }
+
+        QMap<QString, QString> mimeToExt = {
+            {"application/pdf", "pdf"},
+            {"text/plain", "txt"},
+            {"image/png", "png"},
+            {"image/jpeg", "jpg"},
+            {"application/zip", "zip"},
+            {"application/vnd.openxmlformats-officedocument.wordprocessingml.document", "docx"},
+            {"application/msword", "doc"},
+            {"application/vnd.ms-excel", "xls"},
+            {"application/vnd.openxmlformats-officedocument.spreadsheetml.sheet", "xlsx"}
+        };
```

Comment:
Toàn bộ phần mimeToExt nên tách riêng để lên đầu file cpp, hoặc ở một nơi nào đó ===> để sau này dễ quản lý và update các kiểu extension khác